### ISSUE TITLE
Generate Code for a neural network that can be trained on an FPGA

### ISSUE THREAD

**Issue by telefonjoker100** [es-ude/elastic-ai.creator#354]
In this user story the minimal steps for the training on FPGAs are presented:

# General
- [ ] Set up the latest elasticAi.creator version and create a new branch
- [ ] Build a sequential-equivalent module for the training
- [ ] Define the interfaces for the sequential

# Middleware
- [ ] Check if it is necessary to extend the middleware

# Forward neural network

## Linear
- [ ] Reimplement the existing linear layer so parameters are outside of it

## ReLU
- [ ] Check if relu implementation can be used directly

# Gradient Calculation neural network
## Linear
- [ ] Implement the gradient calculation of the linear layer
- [ ] Implement the backward propagation algorithm for the linear layer

## ReLU
- [ ] Implement the backward propagation algorithm for the linear layer

# Optimizer
- [ ] Implement a basic SGD without momentum to update the weights

# Loss-function
- [ ] Implement the MSE loss function
